### PR TITLE
GUI: If BIP70 is disabled, attempt to fall back to BIP21 parsing

### PR DESCRIPTION
**Pull request by luke-jr**

### CODE REVIEW

File: src/qt/paymentserver.cpp [hunk 1]
```diff
@@ -315,19 +315,22 @@ void PaymentServer::handleURIOrFile(const QString& s)
...
                     CClientUIInterface::ICON_WARNING);
             }
-#else
-            Q_EMIT message(tr("URI handling"),
-                tr("Cannot process payment request because BIP70 support was not compiled in."),
-                CClientUIInterface::ICON_WARNING);
-#endif
             return;
         }
+#endif
```
Comment (Empact) [inline src/qt/paymentserver.cpp]: Just from reading this, seems this is invalid when `ENABLE_BIP70` is disabled, due to the unpaired `else` block. Maybe we should disable BIP70 in one of the CI runs?

**Comment by fanquake** [comment]
When I try compiling with `./configure --disable-bip70`:
```
  CXX      qt/libbitcoinqt_a-sendcoinsdialog.o
  CXX      qt/libbitcoinqt_a-sendcoinsentry.o
  CXX      qt/libbitcoinqt_a-signverifymessagedialog.o
qt/paymentserver.cpp:321:9: error: expected expression
        else // normal URI
        ^
1 error generated.
make[2]: *** [qt/libbitcoinqt_a-paymentserver.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [check-recursive] Error 1
make: *** [check-recursive] Error 1
```

**Comment by luke-jr** [comment]
Fixed, added another fix, and a Travis test for no-BIP70

File: .travis.yml [hunk 2]
```diff
@@ -110,6 +110,15 @@ jobs:
         FUNCTIONAL_TESTS_CONFIG="--exclude wallet_multiwallet.py" # Temporarily suppress ASan heap-use-after-free (see issue #14163)
         GOAL="install"
         BITCOIN_CONFIG="--enable-zmq --with-incompatible-bdb --with-gui=qt5 CPPFLAGS=-DDEBUG_LOCKORDER --with-sanitizers=address,integer,undefined CC=clang CXX=clang++"
+# x86_64 Linux, no BIP70
+    - stage: test
+      env: >-
+        HOST=x86_64-unknown-linux-gnu
+        PACKAGES="clang llvm python3-zmq qtbase5-dev qttools5-dev-tools libssl1.0-dev libevent-dev bsdmainutils libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev libdb5.3++-dev libminiupnpc-dev libzmq3-dev libprotobuf-dev protobuf-compiler libqrencode-dev"
```
Comment (fanquake) [inline .travis.yml]: You should be able to skip installing the `libprotobuf-dev` & `protobuf-compiler` packages.

Comment (luke-jr) [reply .travis.yml]: Good point, fixed

**Comment by Empact** [comment]
How about a test case for this behavior?

**Comment by jameshilliard** [comment]
@luke-jr can you rebase this? This should probably be a 0.18 milestone requirement.

**Comment by Sjors** [comment]
cc @MarcoFalke since this also adds one more Travis machine

**Comment by MarcoFalke** [comment]
Don't we already have at least 3 jobs with a gui? I'd prefer to just set the configure flag on one of them.

**Comment by luke-jr** [comment]
@MarcoFalke It doesn't make sense to test this without the wallet. Is there a reason `x86_64 Linux  [GOAL: install]  [xenial]  [no depends, only system libs, sanitizers: thread (TSan), no wallet]` needs to be wallet-less? Or where else would you want to add it?

**Comment by luke-jr** [comment]
(also, rebased)

File: .travis.yml [hunk 3]
```diff
@@ -132,6 +132,15 @@ jobs:
...
         BITCOIN_CONFIG="--enable-zmq --with-incompatible-bdb --with-gui=qt5 CPPFLAGS=-DDEBUG_LOCKORDER --with-sanitizers=address,integer,undefined CC=clang CXX=clang++"
 
+    - stage: test
+      name: 'x86_64 Linux  [GOAL: install]  [no BIP70]'
+      env: >-
+        HOST=x86_64-unknown-linux-gnu
+        PACKAGES="clang llvm python3-zmq qtbase5-dev qttools5-dev-tools libssl1.0-dev libevent-dev bsdmainutils libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev libdb5.3++-dev libminiupnpc-dev libzmq3-dev libqrencode-dev"
+        NO_DEPENDS=1
+        GOAL="install"
+        BITCOIN_CONFIG="--enable-zmq --disable-bip70 --with-incompatible-bdb --with-gui=qt5 CPPFLAGS=-DDEBUG_LOCKORDER --with-sanitizers=address,integer,undefined CC=clang CXX=clang++"
```
Comment (MarcoFalke) [inline .travis.yml]: I'd prefer to set `--disable-bip70` on one of the other bionic jobs that have wallet enabled.

Comment (luke-jr) [reply .travis.yml]: > @MarcoFalke It doesn't make sense to test this without the wallet. Is there a reason `x86_64 Linux [GOAL: install] [xenial] [no depends, only system libs, sanitizers: thread (TSan), no wallet]` needs to be wallet-less? Or where else would you want to add it?

Comment (MarcoFalke) [reply .travis.yml]: Maybe L133, L103, or L93?

Comment (MarcoFalke) [reply .travis.yml]: I think xenial has wallet disabled because the thread sanitizer yells at bdb

**Comment by laanwj** [comment]
utACK 84f53154e1a0309ef582443476451748eb982805

**Comment by Sjors** [comment]
Compiles and `src/qt/test/test_bitcoin-qt` passes on macOS 10.14.3 (`--disable-bip70` & `--enable-bip70`). These tests don't cover much though.

84f5315 looks good modulo Travis, but it seems like a good idea to actually test this. Is there a (testnet or otherwise) site that supports BIP21 fallback for BIP71?

Also can someone upload a BIP70 file with and without such fallback?

**Comment by gmaxwell** [comment]
utACK

**Comment by jonasschnelli** [comment]
utACK 84f53154e1a0309ef582443476451748eb982805

**Pull request merged**